pkg/controllers/managementuser/nodesyncer/cordonfieldssyncer.go

```diff
@@ -275,24 +275,15 @@ func filterErrorMsg(msg string, nodeName string) string {
 	return strings.Join(upd, "\n")
 }
 
-func removeDrainCondition(obj *v3.Node) {
-	exists := false
-	for _, condition := range obj.Status.Conditions {
+func removeDrainCondition(node *v32.Node) {
+	var conditions []v32.NodeCondition
+	for _, condition := range node.Status.Conditions {
 		if condition.Type == "Drained" {
-			exists = true
-			break
-		}
-	}
-	if exists {
-		var conditions []v32.NodeCondition
-		for _, condition := range obj.Status.Conditions {
-			if condition.Type == "Drained" {
-				continue
-			}
-			conditions = append(conditions, condition)
+			continue
 		}
-		obj.Status.Conditions = conditions
+		conditions = append(conditions, condition)
 	}
+	node.Status.Conditions = conditions
 }
```

Comment:
I'm not sure this optimization is safe - there is probably a reason why we first check whether "Drained" condition exists first. With this change, `obj.Status.Conditions` will be formally changed (to different, albeit equivalent) slice which I assume may trigger unnecessary update. Again, I'm not sure if it could lead to any issues but there is non-zero chance of regressions from this unnecessary optimization.
Maybe @kinarashah has more context to share here as this change was done in her PR https://github.com/rancher/rancher/pull/14612.

Reply:
This doesn't seem like an optimization because the drain condition won't always exist and it should just be a noop in those cases. Instead of exiting, it keeps generating a new slice with the same conditions on every sync when node is active. While this might not trigger a cluster update because the conditions remain unchanged, it feels unnecessary compared to the extra check it saves when drain condition is present.

Reply:
Okay, Understood. I'll revert back the changes.